src/expression.js

```diff
@@ -648,8 +648,9 @@ pp.parseIdent = function(liberal) {
   let node = this.startNode()
   if (liberal && this.options.allowReserved == "never") liberal = false
   if (this.type === tt.name) {
-    if (!liberal &&
-        (this.strict ? this.reservedWordsStrict : this.reservedWords).test(this.value) &&
+    let reservedWords = this.options.sourceType === "module" ? this.reservedWordsES6Module :
```

Comment:
In other words, assign `reservedWords` once and for all when initializing parser with given options and `sourceType`. Then this piece will be left intact.